set.go

```diff
@@ -253,3 +253,11 @@ func NewThreadUnsafeSetFromMapKeys[T comparable, V any](val map[T]V) Set[T] {
 
 	return s
 }
+
+func Values[T comparable](s Set[T]) func(func(element T) bool) {
```

Comment:
I still prefer `Elements`, since Set's have elements. Let's make a final decision here and be done with it. As they say: naming things is hard but it's time to move forward on this PR. /cc @amikai